Realm/RLMQueryUtil.mm

```diff
@@ -636,13 +664,13 @@ void add_collection_operation_constraint(NSPredicateOperatorType operatorType,
     if (diacriticSensitive) {
         switch (operatorType) {
             case NSBeginsWithPredicateOperatorType:
-                m_query.and_query(column.begins_with(value, caseSensitive));
+                add_substring_constraint(std::forward<T>(value), column.begins_with(value, caseSensitive));
```

Comment:
Yeah, I thought `value` was a `T&&` for some reason :(. In this case, though, I made a different mistake. I have to pass `value` just as `value`, since moving it here is going to screw up constructing the second parameter (which also uses `value`), and the first argument takes a `const T&` anyways.